Python/dp/README.md

```diff
@@ -1 +1 @@
-a
+[Maximum sum rectangle of 2D array](dp/max_sum.py)
```

Comment:
@shraddhavp Do not add link of code to `Python/dp/README.md` instead add to `Python/README.md` in  `Competitive Programming` section.